user-interface/src/main/java/life/qbic/datamanager/views/general/Tag.java

```diff
@@ -6,11 +6,41 @@ public class Tag extends Span {
 
   public Tag(String text) {
     super(text);
-    addClassName("tag");
+    setClassName("tag");
+    addClassName(TagColor.PRIMARY.getColor());
   }
 
   public Tag(String text, String toolTip) {
     this(text);
     setTitle(toolTip);
   }
+
+  public void setTagColor(TagColor tagColor) {
+    setClassName("tag");
+    addClassName(tagColor.getColor());
```

Comment:
"tag" class should already be set in the constructor?

Reply:
This is a sneaky trick on my part. `setclassName` replaces all current css classes set within the tag with the basic "tag" class. This allows us to remove any previously set color for this tag :)